src/animation/SetStyle.ts

```diff
@@ -0,0 +1,67 @@
+import { Animation } from "./Animation.js";
+import * as THREE from "three";
+import Shape, { Style } from "../geometry/Shape.js";
+
+export default class SetStyle extends Animation {
+  private initialStyle: Style = {};
+  private targetStyle: Style = {};
+
+  constructor(
+    public shape: Shape,
+    public style: Style,
+    config?: any,
+  ) {
+    super(
+      (elapsedTime) => {
+        const currentStyle: Style = {};
+
+        if (this.initialStyle.fillColor && style.fillColor) {
+          currentStyle.fillColor = this.initialStyle.fillColor.clone();
+          currentStyle.fillColor.lerp(style.fillColor, elapsedTime);
+        }
+
+        if (this.initialStyle.strokeColor && style.strokeColor) {
```

Comment:
Is there any reason some of the checks have `style.x !== undefined` but these just have `style.strokeColor/fillColor`?